lib/logstash/filters/translate.rb

```diff
@@ -193,4 +177,58 @@ def filter(event)
       @logger.error("Something went wrong when attempting to translate from dictionary", :exception => e, :field => @field, :event => event)
     end
   end # def filter
+
+  private
+
+  def load_dictionary(registering=false)
+    if /.y[a]?ml$/.match(@dictionary_path)
+      load_yaml(registering)
+    elsif @dictionary_path.end_with?(".json")
+      load_json(registering)
+    elsif @dictionary_path.end_with?(".csv")
+      load_csv(registering)
+    else
+      raise "#{self.class.name}: Dictionary #{@dictionary_path} have a non valid format"
```

Comment:
s/have/has/ . 

Also, this should say that the file extension is invalid. Invalid format sounds like invalid yaml/csv/json syntax.